mypy/subtypes.py

```diff
@@ -390,6 +390,15 @@ def check_type_parameter(
 
 
 class SubtypeVisitor(TypeVisitor[bool]):
+    __slots__ = (
+        "right",
+        "orig_right",
+        "proper_subtype",
+        "subtype_context",
+        "options",
+        "_subtype_kind",
+    )
+
     def __init__(self, right: Type, subtype_context: SubtypeContext, proper_subtype: bool) -> None:
```

Comment:
If a lot of `SubtypeVisitor` objects are created, it follows that a lot of `SubtypeContext` must also be created to pass into it, so I think adding `__slots__` to `SubtypeContext` here should also yield some further improvement.